lychee-bin/src/archive/mod.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::anyhow;
+use std::str::FromStr;
+
+use http::StatusCode;
+use reqwest::{Error, Url};
+use serde::{Deserialize, Deserializer, Serialize};
+
+#[derive(Debug, Serialize, Eq, Hash, PartialEq)]
+pub(crate) struct Suggestion {
+    pub(crate) original: Url,
+    pub(crate) suggestion: Url,
+}
+
+#[derive(Debug, Deserialize, Default, Clone)]
+pub(crate) enum Archive {
+    #[default]
+    WaybackMachine,
+}
+
+impl FromStr for Archive {
+    type Err = anyhow::Error;
+    fn from_str(archive: &str) -> Result<Self, Self::Err> {
+        match archive.to_lowercase().as_str() {
+            "wayback" => Ok(Archive::WaybackMachine),
+            _ => Err(anyhow!("Unknown archive {}", archive)),
+        }
+    }
+}
+
+impl Archive {
+    pub(crate) async fn get_link(&self, original: &Url) -> Result<Option<Url>, Error> {
+        let function = match self {
+            Archive::WaybackMachine => get_wayback_link,
+        };
+
+        function(original).await
+    }
+}
+
+async fn get_wayback_link(url: &Url) -> Result<Option<Url>, Error> {
+    let mut archive_url = Url::parse("https://archive.org/wayback/available").unwrap();
```

Comment:
can we parse this as a static variable? (when it's inside this function, we can clone the static variable and make the clone mutable)

Reply:
Agree. @thomas-zahner, here's how you can do that: https://github.com/lycheeverse/lychee/blob/cd45f9db0745ed58e728984ce4c2d4407582dde8/lychee-lib/src/helpers/path.rs#L9-L10

Reply:
Thank you for the suggestion I did:

```rust
static WAYBACK_URL: Lazy<Url> =
    Lazy::new(|| Url::parse("https://archive.org/wayback/available").unwrap());
```